python_modules/libraries/dagster-dbt/dagster_dbt/dbt_project.py

```diff
@@ -230,5 +236,48 @@ def __init__(
             has_uninstalled_deps=has_uninstalled_deps,
             manifest_preparer=manifest_preparer,
         )
-        if manifest_preparer:
-            manifest_preparer.on_load(self)
+
+    @public
+    def prepared(self) -> "DbtProject":
+        """Execute the preparation process for a dbt project and return the DbtProject object when complete.
```

Comment:
I think it might not be clear to users what "the preparation process" means. Are we able to explain it using other terms? I.e. would it be accurate to say that, if this method returns successfully, it means that `self.manifest_path` will point to a loadable manifest file?

Reply:
Good call - I updated the docstring with more explanation in [6a2a58c](https://github.com/dagster-io/dagster/pull/22513/commits/6a2a58c1764a54c6ac9ba2ce340a68646eaa2c1e)

Reply:
> Do we envision that the preparation process will do things other than create the manifest in the future? If not, might it make sense to just name this something like ensure_manifest, to avoid requiring the user to understand new terminology?

In the current implementation, it runs both `dbt deps`, to pull and install dependencies, and `dbt parse`, to create the manifest. 

Since we plan to have similar classes, like `SdfProject`, I'd advocate using a more general term so we can reuse it in other classes like "the method that makes the project ready for Dagster", if that makes sense.

Also, for the context, we will make the attribute `self.manifest_preparer` experimental before removing the experimental tag on DbtProject, and keep DbtManifestPreparer and DagsterDbtManifestPreparer experimental as well. We'd like to test this with users, as it's likely that some of them would like their preparer to behave differently, eg. run more commands, import the manifest, etc.

It would be nice to have an attribute that fits the method, and use that across the `*Project` classes, eg. `self.preparer` and `prepare()`

@yuhan you mentioned the `SdfProject` use case in the GitHub discussion. Any thoughts on this?

> Separately: aside from the "manifest" vs. "prepare" question, I think that methods with side effects should have names that are verbs.

Thanks for confirming - that makes sense. Something like `prepare()` would work.

> Btw thanks for bearing with me on the nitpick-iness here. More than half of Dagster users use dbt, and the APIs we settle on here will show up during their initial Dagster setup, so I think it's worth doing as much as we can to make sure it's as smooth as possible.

Makes total sense 💯

Reply:
> In the current implementation, it runs both dbt deps, to pull and install dependencies

Do you know why we run dbt deps? Do we need to?

> Since we plan to have similar classes, like SdfProject, I'd advocate using a more general term so we can reuse it in other classes like "the method that makes the project ready for Dagster", if that makes sense.

I suspect most users will probably only use one of these tools. That leads me to lean towards prioritizing using terminology that's specific to the tool they're familiar with, over consistency across different integrations.

Reply:
> Do you know why we run dbt deps? Do we need to?

We handle `dbt deps` in DbtProject since PR #21296 - our dbt NUX was broken for any dbt project that had a `dependencies.yml` file. 

Before DbtProject, the CI/CD files were running both `dbt deps` and `dbt parse`, but now that they use DbtProject, handling `dbt deps` in the preparer was necessary.

Reply:
Got it, handling `dbt deps` in the preparer sounds reasonable to me. Would it be accurate to say though that the reason we run `dbt deps` is so that we can successfully generate the manifest?

Reply:
In the default and current implementation, yes. In this specific case, we need it for `dbt parse`, which creates the manifest.

If a user creates a custom preparer where the manifest is imported, say from a S3 bucket, then no. 

In other words, `dbt deps` is required for other dbt commands to work, like `dbt run` and `dbt build`. If the manifest is provided in any other way then by creating it `dbt parse`, `dbt deps` would still be required.

For these reasons, I'd be in favor of 
1. having a method that is not specifically named after the manifest
2. renaming `DbtManifestPreparer` to `DbtPreparer` and `DagsterDbtManifestPreparer` to `DagsterDbtPreparer`, but I think this is another conversation.

Reply:
The method was renamed from `prepare` to `ensure_prepared`.